libs/agent-sm/agent-policy/src/main/java/org/opensearch/secure_sm/policy/PolicyFile.java

```
@@ -0,0 +1,311 @@
+import java.util.regex.Pattern;
+
+@SuppressWarnings("removal")
+public class PolicyFile extends java.security.Policy {
+
+    private volatile PolicyInfo policyInfo;
+    private URL url;
+
+    /**
+     * When a policy file has a syntax error, the exception code may generate
+     * another permission check and this can cause the policy file to be parsed
+     * repeatedly, leading to a StackOverflowError or ClassCircularityError.
+     * To avoid this, this set is populated with policy files that have been
+     * previously parsed and have syntax errors, so that they can be
+     * subsequently ignored.
+     */
+    private static Set<URL> badPolicyURLs = Collections.newSetFromMap(new ConcurrentHashMap<URL, Boolean>());
+
+    /**
+     * Initializes the Policy object and reads the default policy
+     * from the specified URL only.
+     */
+    public PolicyFile(URL url) {
+        this.url = url;
+        init(url);
+    }
+
+    /**
+     * Initializes the Policy object and reads the default policy
+     * configuration file(s) into the Policy object.
+     *
+     * See the class description for details on the algorithm used to
+     * initialize the Policy object.
+     */
+    private void init(URL url) {
+        PolicyInfo newInfo = new PolicyInfo();
+        initPolicyInfof(url, newInfo);
+        policyInfo = newInfo;
+    }
+
+    private void initPolicyInfof(URL policy, PolicyInfo newInfo) {
+        if (badPolicyURLs.contains(policy)) {
+            return;
+        }
+
+        try (InputStreamReader reader = new InputStreamReader(getInputStream(policy), StandardCharsets.UTF_8)) {
+            PolicyParser policyParser = new PolicyParser();
+            policyParser.read(reader);
+
+            Collections.list(policyParser.grantElements()).forEach(grantNode -> {
+                try {
+                    addGrantNode(grantNode, newInfo);
+                } catch (Exception e) {
+                    e.printStackTrace(System.err);
```

Same applies throughout this method, but do we really want to swallow these exceptions? I would think it would be better to just let the exception propagate to the caller, who can make the decision about how to handle it.

---

Agreed, we should not swallow exceptions. 

I'll clean up the ```PolicyFile.java``` in a new PR revision. My focus was more on the Parser implementation.

---

cleaned up all exception swallow-ing.